src/classes/OpportunityTriggerHandler.cls

```diff
@@ -0,0 +1,474 @@
+                if(opp.StageName == 'Return To HQ'){
+                    for(String hQSensorTags : opp.Return_To_HQ__c.split('\n')){
+                        hQtagSet.add(hQSensorTags.trim());
+                    }
+                    for(String hQSensorTags : hQtagSet){
+                        if(hQSensorTags.length() > 18){
+                            hQSensorTags = hQSensorTags.substring(0,18);
+                        }
+                       
+                        if(hQSensorTags.startsWith('00:11:CE:00') && auSensorOldSet.contains(hQSensorTags)){
+                            for(Automobile_Sensor__c oldAuSensor : oldAuSensorMap.values()){
+                                if(oldAusensor.Name == hQSensorTags &&  oldAuSensor.Opportunity__c == opp.Id){
+                                    oldAuSensor.Account__c = opp.Shipment_Account__c;
+                                    updateAuSensorMap.put(hQSensorTags, oldAuSensor);
+                                }
+                            }
+                        }
+                        if(hQSensorTags.startsWith('00:11:CE:D') && apSensorOldSet.contains(hQSensorTags)){
+                            for(Automobile_Presence_Sensor__c oldApSensor : oldApSensorMap.values()){
+                                if(oldApSensor.Name == hQSensorTags && oldApSensor.Opportunity__c == opp.Id){
+                                    oldApSensor.Account__c = opp.Shipment_Account__c;
+                                    updateApSensorMap.put(hQSensorTags, oldApSensor);
+                                }
+                            }
+                        }   
+                        if(hQSensorTags.startsWith('00:11:CE:E') && aTDUSensorOldSet.contains(hQSensorTags)){
+                            for(Automobile_TDU__c oldATDUSensor : oldATDUSensorMap.values()){
+                                if(oldATDUSensor.Name == hQSensorTags && oldATDUSensor.Opportunity__c == opp.Id){
+                                    oldATDUSensor.Account__c = opp.Shipment_Account__c;
+                                    updateATDUSensorMap.put(hQSensorTags, oldATDUSensor);
+                                }
+                            }
+                        }   
+                        if(hQSensorTags.substring(0,3).isNumeric() && aTagSensorOldSet.contains(hQSensorTags)){
+                            for(Automobile_Tag__c oldATagSensor : oldATagSensorMap.values()){
+                                if(oldATagSensor.Name == hQsensorTags &&  oldATagSensor.Opportunity__c == opp.Id){
+                                    oldATagSensor.Account__c = opp.Shipment_Account__c;
+                                    updateATagSensorMap.put(hQSensorTags, oldATagSensor);
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        
+        //Insert and Update DML Perform
```

Comment:
instead of inserting/updating different lists for different objects what you can do is create the new Sobject list and add all objects lists in one lists update/Insert that SObject list you will save DML